probml_utils/gibbs_finite_mixgauss_utils.py

```diff
@@ -0,0 +1,47 @@
+import jax.numpy as jnp
+from jax import random
+from multivariate_t_utils import log_predic_t
+
+
+def gibbs_gmm(T, X, alpha, K, hyper_params, key):
```

Comment:
Thank you very much for these detailed comments. I'll try to fix these issues and take care in my future code. 
I did find it hard to vectorize the DP mixture model since the sizes of many arrays are not fixed. In fact, even for the finite mixture model, the size of arrays of cluster assignment is not fixed.
As you mentioned, one possible solution is to pre-allocate a large enough vector and then use binary musks, and we might gain time efficiency by sacrificing some space. I'm just a little bit concerned if this approach is still viable when the data is huge.